test/performance/search/search_benchmark.cpp

```diff
@@ -0,0 +1,171 @@
+
+using namespace seqan3;
+
+template <typename alphabet_t>
+auto generate_sequence_seqan3(size_t const len = 500,
+                              size_t const variance = 0,
+                              size_t const seed = 0)
+{
+    std::mt19937 gen(seed);
+    std::uniform_int_distribution<uint8_t> dis_alpha(0, alphabet_size_v<alphabet_t> - 1);
+    std::uniform_int_distribution<size_t> dis_length(len - variance, len + variance);
+
+    std::vector<alphabet_t> sequence;
+
+    size_t length = dis_length(gen);
+    for (size_t l = 0; l < length; ++l)
+        sequence.push_back(alphabet_t{}.assign_rank(dis_alpha(gen)));
+
+    return sequence;
+}
+
+template<typename alphabet_t>
+void mutateInsertion(std::vector<alphabet_t> & seq, size_t const  errors){
+    auto const alphabetSize = alphabet_t::value_size;
+    uint32_t rpos = rand() % (seq.size() - errors);
+    int rValue = rand() % alphabetSize;
+    alphabet_t cbase;
+    cbase.assign_rank(rValue);
+    seq.insert(seq.begin() + rpos, cbase);
+}
+
+template<typename alphabet_t>
+void mutateDeletion(std::vector<alphabet_t> & seq, size_t const  errors){
+    uint32_t rpos = rand() % (seq.size() - errors);
+    seq.erase(seq.begin() + rpos);
+}
+
+template<typename alphabet_t>
+void mutateSubstitution(std::vector<alphabet_t> & seq, size_t const  errors){
+    auto const alphabetSize = alphabet_t::value_size;
+    uint32_t rpos = rand() % (seq.size() - errors);
+    int rValue = rand() % (alphabetSize - 1);
+    alphabet_t & cbase = seq[rpos];
+    int cValue = to_rank(cbase);
+    if(rValue >=  cValue)
+        ++rValue;
+    cbase.assign_rank(rValue);
+}
+
+template<typename alphabet_t>
+void generate_reads(std::vector<std::vector<alphabet_t> > & reads,
+                    std::vector<alphabet_t> & ref,
+                    size_t const number_of_Reads,
+                    size_t const read_length,
+                    size_t const simulated_errors,
+                    float const probInsertion,
+                    float const probDeletion
+                   )
+{
+    for(size_t i = 0; i < number_of_Reads; ++i){
+        size_t rpos = rand() % (ref.size() - read_length- simulated_errors);
+        std::vector<alphabet_t> query_tmp{ref.begin() + rpos, ref.begin() + rpos + read_length + simulated_errors};
+        for(size_t j = 0; j < simulated_errors; ++j)
+        {
+            //Substitution
+            float prob = (float) rand()/RAND_MAX;
+            if(probInsertion + probDeletion < prob)
+            {
+                mutateSubstitution(query_tmp, simulated_errors);
+            }
+            //Insertion
+            else if(probInsertion < prob)
+            {
+                mutateInsertion(query_tmp, simulated_errors);
+            }
+            //Deletion
+            else
+            {
+                mutateDeletion(query_tmp, simulated_errors);
+            }
+        }
+        std::vector<alphabet_t>  read{query_tmp.begin(), query_tmp.begin() + read_length};
```

Comment:
You could make the code only copy once from the ref to the read. But it's fine for benchmarking :)